hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/ConnectionManager.java

```diff
@@ -229,7 +229,7 @@ public ConnectionContext getConnection(UserGroupInformation ugi,
 
     // Add a new connection to the pool if it wasn't usable
     if (conn == null || !conn.isUsable()) {
-      if (!this.creatorQueue.offer(pool)) {
+      if (!this.creatorQueue.contains(pool) && !this.creatorQueue.offer(pool)) {
```

Comment:
Sorry, I don’t understand the meaning of this change. Can we explain the reason for this change?

Reply:
Prevents duplicate pool from being added to the creatorQueue.   getConnection() will be concurrent called, so createQueue will be added duplicate pool.